webApp/src/main/java/com/infoshare/controller/VolunteerController.java

```diff
@@ -20,7 +25,25 @@ public VolunteerController(VolunteerService volunteerService) {
 
     @GetMapping("/create")
     public String createVolunteer(Model model) {
-        return getTestViewWithPageUnderConstructionMessage(model);
+        model.addAttribute(new VolunteerForm());
+        model.addAttribute("types",volunteerService.getTypesOfHelp());
+        return "volunteer-register-form";
+    }
+    @PostMapping("/form-details")
+    public String createVolunteerFormDetails( @ModelAttribute("volunteerForm") VolunteerForm volunteerFrom, BindingResult br, Model model) {
+        if (br.hasErrors()) {
+            model.addAttribute("types", volunteerService.getTypesOfHelp());
+            return "/volunteer-register-form";
+        } else {
+            volunteerService.registerNewVolunteer(volunteerFrom.getName(), volunteerFrom.getLocation(), volunteerFrom.getEmail(),
+                    volunteerFrom.getPhone(), volunteerFrom.getTypeOfHelp(), volunteerFrom.isAvalible());
+            return "redicted:/volunteer/all";
```

Comment:
Tutaj jest literowka w wyrazie, powino byc: 
```suggestion
            return "redirect:/volunteer/all";
```

Reply:
poprawione